### DISCUSSION TITLE
Permission Hierarchy Issue

### DISCUSSION THREAD

**Discussion by rlarrode** [requarks/wiki#3517] [Error / Bug Report]
**Describe the bug**
I try to manage access permission using group permissions

I have to groups:
members
advanced

I have one page: 
name: mypage
path: /mypath/mypage
tags: [mytag]

I setup permissions as follow:
deny members for path start /mypath
grant advanced for tags match mytag

I give my user both groups at the same time

He cannot access the page

**To Reproduce**
See description

**Expected behavior**
User should see the page as `tag match` rule should be higher than `path starts with` as per documentation

**Host Info (please complete the following information):**
 - OS: Debian 9.10
 - Wiki.js version: 2.3.81
 - Database engine: postgres 9.5.20
- Nodejs: 12.16.3

**Additional context**
i analyzed the server/core/auth.js file / function _applyPageRuleSpecificity
```
 _applyPageRuleSpecificity ({ rule, checkState, higherPriority = [] }) {
    if (rule.path.length === checkState.specificity.length) { //////Are we only checking the number of characters and not that the strings matches//////
      // Do not override higher priority rules
      if (_.includes(higherPriority, checkState.match)) { 
        return checkState
      }
      // Do not override a previous DENY rule with same match
      if (rule.match === checkState.match && checkState.deny && !rule.deny) {
        return checkState
      }
    } else if (rule.path.length < checkState.specificity.length) {//////Are we only checking the number of characters and not that the strings matches? If yes then tags will mostly be lower thant paths !!!!//////
      // Do not override higher specificity rules
      return checkState
    }

    return {
      deny: rule.deny,
      match: rule.match,
      specificity: rule.path
    }
  }
```
Thanks for your feedback

**Comment by ba-moun** [comment]
Same problem in Wiki.js v2.4.105.

**Comment by ba-moun** [comment]
Hello @NGPixel,
Does this issue will be fixed soon?
